torchgeo/datasets/rwanda_field_boundary.py

```diff
@@ -0,0 +1,328 @@
+                        f"nasa_rwanda_field_boundary_competition_source_{split}",
+                        f"nasa_rwanda_field_boundary_competition_source_{split}_{i:02d}_{date}",  # noqa: E501
+                        f"{band}.tif",
+                    )
+                    patch.append(fn)
+                dates.append(patch)
+            self.image_filenames.append(dates)
+            self.mask_filenames.append(
+                os.path.join(
+                    self.root,
+                    f"nasa_rwanda_field_boundary_competition_labels_{split}",
+                    f"nasa_rwanda_field_boundary_competition_labels_{split}_{i:02d}",
+                    "raster_labels.tif",
+                )
+            )
+
+    def __getitem__(self, index: int) -> dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            a dict containing image, mask, transform, crs, and metadata at index.
+        """
+        img_fns = self.image_filenames[index]
+        mask_fn = self.mask_filenames[index]
+
+        imgs = []
+        for date_fns in img_fns:
+            bands = []
+            for band_fn in date_fns:
+                with rasterio.open(band_fn) as f:
+                    bands.append(f.read(1).astype(np.int32))
+            imgs.append(bands)
+        img = torch.from_numpy(np.array(imgs))
+
+        sample = {"image": img}
+
+        if self.split == "train":
+            with rasterio.open(mask_fn) as f:
+                mask = f.read(1)
+            mask = torch.from_numpy(mask)
+            sample["mask"] = mask
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def __len__(self) -> int:
+        """Return the number of chips in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.image_filenames)
+
+    def _validate_bands(self, bands: Sequence[str]) -> None:
+        """Validate list of bands.
+
+        Args:
+            bands: user-provided sequence of bands to load
+
+        Raises:
+            ValueError: if an invalid band name is provided
+        """
+        for band in bands:
+            if band not in self.all_bands:
+                raise ValueError(f"'{band}' is an invalid band name.")
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset.
+
+        Raises:
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+        """
+        # Check if the subdirectories already exist and have the correct number of files
+        checks = []
+        for split, num_patches in self.number_of_patches_per_split.items():
+            path = os.path.join(
+                self.root, f"nasa_rwanda_field_boundary_competition_source_{split}"
+            )
+            if os.path.exists(path):
+                num_files = len(os.listdir(path))
```

Comment:
This could get buggy, for example, if you load one of the images in QGIS and it results in a hidden file left behind